core/src/main/java/de/danoeh/antennapod/core/service/playback/PlaybackService.java

```diff
@@ -497,6 +497,7 @@ public int onStartCommand(Intent intent, int flags, int startId) {
                     return Service.START_NOT_STICKY;
                 }
                 mediaPlayer.playMediaObject(playable, stream, startWhenPrepared, prepareImmediately);
+                AddPlayableToQueue(playable);
```

Comment:
Of the 4 cases where playMediaObject() is called, this was the only case I was able to trigger when testing - how do I call OnPlayFromSearch(), OnPlayFromMediaId(), and StartPlayingFromPreferences()?

Reply:
Thanks for your pull request. I like the fact that you added a test and try to increase the test coverage :+1:

`StartPlayingFromPreferences` can be triggered by sending a Play button keycode to the service without specifying a media file (see `PlaybackServiceNotificationBuilder.getPendingIntentForMediaAction`).

The other methods are related to Android Auto, I think. Not sure how to test those.

Reply:
It's hard to write tests for the other cases since they require a lot of internal states within the app to be configured appropriately, so I think this is the only test I'm going to add for now. Is that ok/do you think the PR can be merged in?

Reply:
> Is that ok/do you think the PR can be merged in?

Totally ok :) I did not expect a test in the first place, so even a single one is really nice to have :+1:

I will have a closer look at the PR soon.

Reply:
@ByteHamster Have you gotten a chance to look over the pr?